### PR TITLE
use mini_magick and mvg file

### PR DESCRIPTION
**Pull request by pedrocarmona**
Use mini_magick to avoid memory leaks. 
Use mvg file to simplify generation, using ERB template.

### CODE REVIEW

File: lib/worldize.rb [hunk 1]
```diff
@@ -40,39 +40,58 @@ def country_names
...
-        bg = countries_and_options[country.properties.name] ||
-              countries_and_options[country.properties.iso_a3] ||
-              options.fetch(:land)
-        draw_country(gc, country, width, bg)
-        gc.fill(options.fetch(:land))
+    def template
+      template_filename = 'worldize/template.mvg.erb'
+      template_path = File.join(File.dirname(__FILE__), template_filename)
+      ERB.new(File.read(template_path), nil, '-')
+    end
```
Comment (zverok) [inline lib/worldize.rb:47]: Constant(s)?

File: lib/worldize.rb [hunk 2]
```diff
@@ -40,39 +40,58 @@ def country_names
...
+
+    # MiniMagick cannot ha
```
Comment (zverok) [inline lib/worldize.rb:49]: Unfinished comment?

File: lib/worldize.rb [hunk 3]
```diff
@@ -40,39 +40,58 @@ def country_names
...
+      file = Tempfile.new(['worldize', '.mvg'])
+      begin
+        mvg_content = template.result(binding)
+        file.write(mvg_content)
+        file.close
+        png = convert_mvg_to_png(file, width, ymin, ymax)
+      ensure
+        file.close
+        file.unlink # deletes the temp file
+      end
```
Comment (zverok) [inline lib/worldize.rb:88]: I believe the code in lines 79-88 is identical to
```ruby
Tempfile.create(['worldize', '.mvg']) do |file|
  file.write(template.result(binding))
  file.close
  convert_mvg_to_png(file, width, ymin, ymax)
end
```

File: lib/worldize.rb [hunk 4]
```diff
@@ -40,39 +40,58 @@ def country_names
...
-        gc.fill(options.fetch(:land))
+    def template
+      template_filename = 'worldize/template.mvg.erb'
+      template_path = File.join(File.dirname(__FILE__), template_filename)
+      ERB.new(File.read(template_path), nil, '-')
+    end
+
+    # MiniMagick cannot ha
+    # rubocop:disable Metrics/MethodLength
+    def convert_mvg_to_png(input_file, width, ymin, ymax)
```
Comment (zverok) [inline lib/worldize.rb:51]: just `mvg_to_png`?

File: lib/worldize.rb [hunk 5]
```diff
@@ -40,39 +40,58 @@ def country_names
...
+          convert << temp.path
+        end
+        png = MiniMagick::Image.open(temp.path, '.png')
+        png.combine_options do |c|
+          c.crop "#{width}x#{ymax - ymin}+0+#{ymin}!"
+        end
+      ensure
+        temp.close
+        temp.unlink # deletes the temp file
       end
```
Comment (zverok) [inline lib/worldize.rb:66]: I believe in this code you are just using the name that `Tempfile` library have provided?.. Then probably it is better to just do
```ruby
temp_path = Dir::Tmpname.create(%w[worldize .png]) {} # yes, it has a bit weird call requirements with mandatory block
```

File: lib/worldize.rb [hunk 6]
```diff
@@ -40,39 +40,58 @@ def country_names
...
+      template_path = File.join(File.dirname(__FILE__), template_filename)
+      ERB.new(File.read(template_path), nil, '-')
+    end
+
+    # MiniMagick cannot ha
+    # rubocop:disable Metrics/MethodLength
+    def convert_mvg_to_png(input_file, width, ymin, ymax)
+      temp = Tempfile.new(['worldize', '.png'])
+      begin
+        # workaround, there isnt a way to enforce `mvg:filename` decode
```
Comment (zverok) [inline lib/worldize.rb:54]: I don't understand this comment, can you please explain? :(

File: lib/worldize.rb [hunk 7]
```diff
@@ -40,39 +40,58 @@ def country_names
...
+        MiniMagick::Tool::Convert.new do |convert|
+          convert << "mvg:#{input_file.path}"
+          convert << temp.path
+        end
+        png = MiniMagick::Image.open(temp.path, '.png')
+        png.combine_options do |c|
+          c.crop "#{width}x#{ymax - ymin}+0+#{ymin}!"
+        end
```
Comment (zverok) [inline lib/worldize.rb:62]: Will not this work, instead?
```ruby
MiniMagick::Tool::Convert.new do |convert|
  convert << "mvg:#{input_file.path}" << temp.path
  convert.crop "#{width}x#{ymax - ymin}+0+#{ymin}!"
end
```

**Review by zverok — COMMENTED** [review]
Sorry for long time before review, got distracted by some personal matters.

I generally love the idea, left some comments in code.

**Comment by pedrocarmona** [comment]
Sorry, I end up not using it since, so going to close, Thank you

**Pull request closed without merging**